tests/components/blebox/test_switch.py

```diff
@@ -0,0 +1,386 @@
+
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_OFF, {"entity_id": entity_id}, blocking=True,
+    )
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+
+def relay_mock(relay_id=0):
+    """Return a default switchBoxD switch entity mock."""
+
+    return mock_only_feature(
+        blebox_uniapi.switch.Switch,
+        unique_id=f"BleBox-switchBoxD-1afe34e750b8-{relay_id}.relay",
+        full_name=f"switchBoxD-{relay_id}.relay",
+        device_class="relay",
+        is_on=None,
+    )
+
+
+@pytest.fixture(name="switchbox_d")
+def switchbox_d_fixture():
+    """Set up two mocked Switch features representing a switchBoxD."""
+
+    relay1 = relay_mock(0)
+    relay2 = relay_mock(1)
+    features = [relay1, relay2]
+
+    product = setup_product_mock("switches", features)
+
+    type(product).name = PropertyMock(return_value="My relays")
+    type(product).model = PropertyMock(return_value="switchBoxD")
+    type(product).brand = PropertyMock(return_value="BleBox")
+    type(product).firmware_version = PropertyMock(return_value="1.23")
+    type(product).unique_id = PropertyMock(return_value="abcd0123ef5678")
+
+    type(relay1).product = product
+    type(relay2).product = product
+
+    return (features, ["switch.switchboxd_0_relay", "switch.switchboxd_1_relay"])
+
+
+async def test_switchbox_d_init(switchbox_d, hass, config):
+    """Test switch default state."""
+
+    feature_mocks, entity_ids = switchbox_d
+
+    feature_mocks[0].async_update = AsyncMock()
+    feature_mocks[1].async_update = AsyncMock()
+    entries = await async_setup_entities(hass, config, entity_ids)
+
+    entry = entries[0]
+    assert entry.unique_id == "BleBox-switchBoxD-1afe34e750b8-0.relay"
+
+    state = hass.states.get(entity_ids[0])
+    assert state.name == "switchBoxD-0.relay"
+    assert state.attributes[ATTR_DEVICE_CLASS] == DEVICE_CLASS_SWITCH
+    assert state.state == STATE_OFF  # NOTE: should instead be STATE_UNKNOWN?
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My relays"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "switchBoxD"
+    assert device.sw_version == "1.23"
+
+    entry = entries[1]
+    assert entry.unique_id == "BleBox-switchBoxD-1afe34e750b8-1.relay"
+
+    state = hass.states.get(entity_ids[1])
+    assert state.name == "switchBoxD-1.relay"
+    assert state.attributes[ATTR_DEVICE_CLASS] == DEVICE_CLASS_SWITCH
+    assert state.state == STATE_OFF  # NOTE: should instead be STATE_UNKNOWN?
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My relays"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)